week2/week2-2/index.js

```diff
@@ -0,0 +1,199 @@
+const INIT_BALANCE = 0
+const HISTORY_LIST = [
+  {
+    category: '식비',
+    location: '버거킹 고양스타필드점',
+    price: 10_800,
+    type: '지출',
+  },
+  {
+    category: '취미',
+    location: '더클라임 강남점',
+    price: 23_000,
+    type: '지출',
+  },
+  {
+    category: '월급',
+    location: 'CU 고양원당점',
+    price: 300_000,
+    type: '수입',
+  },
+  {
+    category: '쇼핑',
+    location: '쿠팡',
+    price: 4_200,
+    type: '수입',
+  },
+]
+
+const TOTAL_HISTORY = document.querySelector('.detail__list');
+
+const TOTAL_BALANCE = document.querySelector('.total__sum__amount');
+const TOTAL_INCOME = document.querySelector('.total__detail__item__plus__num');
+const TOTAL_SPENDING = document.querySelector('.total__detail__item__minus__num');
+const DELETE_BUTTON = document.querySelector('.detail__list__item__delete');
+const ADD_BUTTON = document.querySelector('.footer__btn');
+
+const INCOME_CHECK_BOX = document.getElementById('income');
+const SPENDING_CHECK_BOX = document.getElementById('spending');
+const CLOSE_BUTTON = document.getElementById('close-modal');
+const INCOME_RADIO = document.getElementById('add-income');
+const SPENDING_RADIO = document.getElementById('add-spending');
+const INCOME_SELECT = document.getElementById('select-income');
+const SPENDING_SELECT = document.getElementById('select-spending');
+const SAVE_INFO_BUTTON = document.getElementById('save-info');
+
+let totalBalance = INIT_BALANCE; // 굳이 왜 이렇게?
+let totalIncome = 0;
+let totalSpending = 0;
+
+const updateHistory = function updateHistoryListWithCheckBox() {
+  const SHOW_INCOME = INCOME_CHECK_BOX.checked;
+  const SHOW_SPENDING = SPENDING_CHECK_BOX.checked;
+  
+  const EACH_HISTORY = document.querySelectorAll('.detail__list__item');
+  
+  EACH_HISTORY.forEach((item) => {
+    const SELECTED_TYPE = item.querySelector('.detail__list__item__history__price').classList.contains('income')
+    ? '수입'
+    : '지출';
+    
+    if ((SHOW_INCOME && SELECTED_TYPE === '수입') || (SHOW_SPENDING && SELECTED_TYPE === '지출')) {
+      item.style.display = 'flex';
+    } else {
+      item.style.display = 'none';
+    }
+  });
+}
+
+const useIncomeSelect = function updateSelectByType() {
+  INCOME_SELECT.classList.add('add-list__input__selected');
+  SPENDING_SELECT.classList.remove('add-list__input__selected');
+}
+
+const useSpendingSelect = function updateSelectByType() {
+  INCOME_SELECT.classList.remove('add-list__input__selected');
+  SPENDING_SELECT.classList.add('add-list__input__selected');
+}
```

Comment:
`toggle`을 사용하면 하나의 로직으로 묶을 수 있을 것 같아요!
```suggestion
const useTypeSelect = function updateSelectByType() {
  INCOME_SELECT.classList.toggle('add-list__input__selected');
  SPENDING_SELECT.classList.toggle('add-list__input__selected');
}
```